android/src/main/java/com/ryanheise/just_audio/MethodCallHandlerImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.ryanheise.just_audio;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import io.flutter.plugin.common.BinaryMessenger;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MethodCallHandlerImpl implements MethodCallHandler {
+
+	private final Context applicationContext;
+	private final BinaryMessenger messenger;
+
+	private final List<AudioPlayer> players = new ArrayList<>();
+
+	public MethodCallHandlerImpl(Context applicationContext,
+			BinaryMessenger messenger) {
+		this.applicationContext = applicationContext;
+		this.messenger = messenger;
+	}
+
+	@Override
+	public void onMethodCall(MethodCall call, @NonNull Result result) {
+		switch (call.method) {
+			case "init":
+				final List<?> args = (List<?>) call.arguments;
+				String id = (String) args.get(0);
+				players.add(new AudioPlayer(applicationContext, messenger, id));
+				result.success(null);
+				break;
+			case "setIosCategory":
+				result.success(null);
+				break;
+			default:
+				result.notImplemented();
+				break;
+		}
+	}
+
+	void dispose() {
+		for (AudioPlayer player : players) {
+			player.dispose();
+		}
+
+		players.clear();
```

Comment:
I understand the benefits of maintaining this list of players, although it appears that this list grows ever larger until the flutter engine is destroyed. Ideally player instances would be removed from this list as soon as an app calls `player.dispose()` from the Flutter side. It is perhaps not as devastating as it may otherwise be to leave these players in memory since just its just the shell that remains in memory, while the heavyweight ExoPlayer kernel is at least disposed of. It is tempting to handle "dispose" on the main channel here where you would have access to the players list but that would require a change to the other platform implementations + testing. A more isolated way is to pass a reference to `this` into the new player which it can later call back on during `dispose` to do this cleanup.

If it sounds not worth it, don't worry, you can leave it as is and I might look into this again later.

Reply:
How about moving the dispose method to the MethodCallHandlerImpl, where it can manage the removal of the list item and calls the dispose method on the AudioPlayer object from the outside?

Reply:
That's what I meant by "It is tempting to handle dispose ... here" in my comment above. But this method handler class (which should probably be named in a less anonymous way - `MainMethodCallHandler`?) is for the main method channel and moving this method from one channel to another will impact the other platform implementations (iOS and web). I'd prefer to avoid making a change with such impact unless it's fully thought through, so I'd prefer for now something that only affects the Android implementation.

Reply:
Oh, i see you already did that. But you see, this will break the iOS and web implementations - isolating this to just the Android implementation will be better for now, and the bigger picture can be looked at afterwards.

Reply:
@ryanheise ok, let me sort it and update the PR again.

Reply:
@ryanheise updated once again.

Reply:
Great, assuming that's good I think it's ready to merge, but it's 3:50am so I'm going to sleep and will take a fresh look in the morning :-)